src/socket/web/server.ts

```diff
@@ -44,23 +47,23 @@ export class MuWebSocketConnection {
     public onClose:() => void = noop;
     public serverClose:() => void;
 
-    constructor (sessionId:string, reliableSocket:WSSocket, serverClose:() => void) {
+    constructor (sessionId:string, reliableSocket:WSSocket, serverClose:() => void, logger:MuLogger) {
         this.sessionId = sessionId;
         this.reliableSocket = reliableSocket;
         this.serverClose = serverClose;
+        this._logger = logger;
 
         this.reliableSocket.onmessage = ({ data }) => {
+            if (this.closed) {
+                return;
+            }
             if (this.started) {
-                if (typeof data === 'string') {
-                    this.onMessage(data, false);
-                } else {
-                    this.onMessage(new Uint8Array(data), false);
-                }
+                this.onMessage(data, false);
             } else {
                 if (typeof data === 'string') {
                     this.pendingMessages.push(data);
                 } else {
-                    this.pendingMessages.push(new Uint8Array(data).slice(0));
```

Comment:
i remember adding the duplication to solve some bug, but i don't remember what kind of bug

Reply:
that was the bug i solved.